src/app/admin/mod.rs

```diff
@@ -3,32 +3,41 @@
 //! * `/metrics` -- reports prometheus-formatted metrics.
 //! * `/ready` -- returns 200 when the proxy is ready to participate in meshed traffic.
 
-use futures::future::{self, FutureResult};
+use futures::future::{self, Future, FutureResult};
 use http::StatusCode;
 use hyper::{service::Service, Body, Request, Response};
 use std::io;
 
 use metrics;
 
 mod readiness;
+mod trace_level;
 pub use self::readiness::{Latch, Readiness};
+use self::trace_level::TraceLevel;
 
 #[derive(Debug, Clone)]
 pub struct Admin<M>
 where
     M: metrics::FmtMetrics,
 {
     metrics: metrics::Serve<M>,
+    trace_level: TraceLevel,
     ready: Readiness,
 }
 
+pub type ResponseFuture = future::Either<
+    FutureResult<Response<Body>, io::Error>,
+    Box<Future<Item = Response<Body>, Error = io::Error> + Send + 'static>,
+>;
```

Comment:
Do you think we should just make this all `Box<...>` without the either? I'm not particularly worried about boxing metrics responses...

Reply:
I'm fine with boxing them all, my first pass at this branch did that. I just thought it would be trivial to avoid the box in the `/metrics` and `/ready` cases, so i went ahead and did it. happy to change it back.

Reply:
we're clearly headed down the path of this signature getting more complex so i'd just simplify it to Box for now.